vllm/v1/worker/gpu_input_batch.py

```diff
@@ -192,6 +192,9 @@ def __init__(
 
         self.logit_bias: List[Optional[Dict[int,
                                             float]]] = [None] * max_num_reqs
+        self.has_allowed_token_ids: List[bool] = [False] * max_num_reqs
```

Comment:
Similar to the other params, better for this to be a set of request indices. Then `no_allowed_token_ids` can just check whether the set is empty without having to iterate, and in `remove_request`, if the index was not already in the set, no need to clear that row of the mask.

Reply:
yeah, switched to a set